crates/wasm-encoder/src/reencode.rs

```diff
@@ -0,0 +1,1308 @@
+#![allow(missing_docs)] // FIXME
+
+#[derive(Debug)]
+pub enum Error {
+    /// There was a type reference that was canonicalized and no longer
+    /// references an index into a module's types space, so we cannot encode it
+    /// into a Wasm binary again.
+    CanonicalizedHeapTypeReference,
+    /// The const expression is invalid: not actually constant or something like
+    /// that.
+    InvalidConstExpr,
+    /// There was an error when parsing.
+    ParseError(wasmparser::BinaryReaderError),
```

Comment:
This'll probably want a `From` implementation to avoid all the `.map_err(Error::ParseError)` below too

Reply:
Done